src/Tribe/Aggregator/Record/Queue_Cleaner.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+
+class Tribe__Events__Aggregator__Record__Queue_Cleaner {
+
+	/**
+	 * @var int The time a record is allowed to stall before having
+	 *          its status set to failed in seconds.
+	 */
+	protected $stall_limit = HOUR_IN_SECONDS;
+
+	/**
+	 * Removes duplicate records for the same import ID.
+	 *
+	 * While it makes sense to keep track of past import records it does not make sense
+	 * to keep more than one pending record for the same import ID.
+	 *
+	 * @param Tribe__Events__Aggregator__Record__Abstract $record A record object or a record post ID.
+	 *
+	 * @return int[] An array containing the deleted posts IDs.
+	 */
+	public function remove_duplicate_pending_records_for( Tribe__Events__Aggregator__Record__Abstract $record ) {
+		if ( empty( $record->meta['import_id'] ) ) {
+			return array();
+		}
+
+		$import_id = $record->meta['import_id'];
+
+		/** @var \wpdb $wpdb */
+		global $wpdb;
+
+		$pending_status = Tribe__Events__Aggregator__Records::$status->pending;
+
+		$query = $wpdb->prepare( "SELECT ID
+			FROM {$wpdb->postmeta} pm
+			JOIN {$wpdb->posts} p
+			ON pm.post_id = p.ID
+			WHERE p.post_type = %s
+			AND p.post_status = %s
+			AND pm.meta_key = '_tribe_aggregator_import_id'
+			AND pm.meta_value = %s
+			ORDER BY p.post_modified_gmt DESC", Tribe__Events__Aggregator__Records::$post_type, $pending_status, $import_id );
```

Comment:
the `WP_Query` is slow and prone to error due to indiscriminate third-party intervention. Here we are cleaning home with a non front-facing post type; I would allow someone to mess with this logic and query if there really is a request for it.